shim/index.js

```diff
@@ -16,14 +16,21 @@ module.exports = async function run(action) {
     }
 
     // Check for version number
-    if (process.env.GITHUB_ACTION_REF && process.env.GITHUB_ACTION_REF.startsWith("v") > 0) {
-        version = process.env.GITHUB_ACTION_REF;
+    if (process.env.GITHUB_ACTION_REF) {
+        if (process.env.GITHUB_ACTION_REF.startsWith("v")) {
+            version = process.env.GITHUB_ACTION_REF;
+        } else if (process.env.GITHUB_ACTION_REF !== "master") {
+            throw new Error(`Invalid version: ${process.env.GITHUB_ACTION_REF}`)
+        }
     }
+
+
     core.info(`Using version ${version}`)
 
     let toolPath;
     // Download the binary if not in local mode
     if (!isLocalMode) {
+        const cacheVersion = `${semver.coerce(version).version}-${process.env.GITHUB_RUN_ID}-${process.env.GITHUB_RUN_ATTEMPT}`;
```

Comment:
I think there should be a comment explaining why we are doing this, otherwise it would seem very odd